### DISCUSSION TITLE
Code block / raw text execute and embed results

### DISCUSSION THREAD

**Discussion by IllustratedMan-code** [typst/typst#899] [Ideas]
This might be a bit beyond the scope of this project, but here goes:

I'm coming from the Emacs world, which has the (somewhat unique) ability to run code blocks and output the results directly into the document through a system called [org-babel](https://orgmode.org/worg/org-contrib/babel/).  This is a feature I use every day, and I think it would be a great addition to typst! 

> note: Jupyter notebooks have a somewhat equivalent feature set, but the implementation is less flexible.

Org-babel works by calling an existing program on the host computer. It then takes the results and either outputs them as is, or does some operation on them (convert to table).

An example org mode document with a code block might work like so:
```
#+begin_src python
print("hello, world")
#+end_src python
#+RESULTS:
: hello world
```

It should be noted that org babel can wrap the function with whatever, say a c++ code block could be wrapped in an `int main(){}` .function. So there would have to be some way of defining these wrappers. Code blocks can also make use of extra header arguments to adjust how this works. 

I'm not entirely sure of the best way of adding this to typst, but maybe the compiler could pass the code blocks to some external process, and or have some interface that an editor could hook into. 

I would love to implement this myself, but would like some pointers and or opinions about this idea before I start.

**Comment by nahuel** [comment]
Maybe something like this? https://typst.app/universe/package/pyrunner/ 
Also check https://forum.typst.app/t/how-to-call-external-applications/1607
